[10.0] Add hr_holidays_jours_ouvrables

guewen opened the pull request:
This module allows to handle French leaves based on "Jours Ouvrables".
Not knowing how to translate that, I kept the French words. If you have a better proposal...

Commit:
Add hr_holidays_jours_ouvrables

Commit:
Add french translation for hr_holidays_jours_ouvrables

Commit:
Fix user permission

Managers must be able to edit saturdays (not yet implemented).
Users should not, so they should be created with sudo().
Complete the tests so leaves are created with a employee user, without
manager access.

Commit:
Add checks on Saturdays bounds

Commit:
Improve wording

sebalix commented:
Maybe "working days"?
We already started a  similar module for an old 6.0 project and the code was based on https://pypi.python.org/pypi/workdays/ so I think this is the right term.

guewen commented:
> Maybe "working days"?

Isn't it equivocal with the "Jours ouvrés"? If you tell me "working days", I would think about "jours ouvrés".

sebalix commented:
Yes you are right, I mixed them up.

Commit:
Add OCA dependency on HR

Commit:
Try to prevent error on Travis

I could not reproduce the bug on my side, but happens on Travis.

File "/home/travis/build/OCA/l10n-france/hr_holidays_jours_ouvrables/models/hr_holidays.py", line 44, in _compute_number_of_days
    days = days + len(saturdays)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

Checking if the tests pass with this quick fix.

gurneyalex commented:
@guewen needs rebase

guewen commented:
Rebase done, conflict resolved (in oca_dependencies.txt).

gurneyalex commented:
@guewen I rebased again, but travis was red...

Commit:
Fix method name changed in dependency

Changed in: https://github.com/OCA/hr/commit/4b155ea

Commit:
Fix new flake8 errors (E722 do not use bare except)

sbidoul commented on account_banking_fr_lcr/models/account_payment_order.py:
```diff
@@ -31,7 +31,9 @@ def _prepare_field(self, field_name, field_value, size):
             for unallowed_ascii_char in unallowed_ascii_chars:
                 value = value.replace(unallowed_ascii_char, '-')
-        except:
+        except BaseException:
```
In my understanding, in most case it should be `Exception` instead of `BaseException`, because `BaseException` also includes system exiting exceptions.

guewen replied:
@sbidoul that's what I thought too but @yvaucher showed me that `python -m autopep8 --select=E722 -a test.py` replace `except:` by `except BaseException:`
And `except Exception` still raises E722

guewen replied:
I read something wrong without checking :) `except Exception` does not raise E722 hopefully, I fixed that.

yvaucher replied:
It appears we cheched with @guewen that E722 isn't raised with `except Exception:` so it the solution to go.

I think that autopep8 does it to make it exactly the same as `except:` but it is a bad idea in most case to keep it that way for the reason @sbidoul explained.

gurneyalex commented:
Hello @guewen I tested this on runbot and it looks good to me. Just 1 remark:

* to set the company on the leave type, you need to be in the multicompany group. Maybe this needs to be noted in the readme (non blocking)

Also if others want to test this, remember to set the work address of the employee to France (or the same country where you define public holidays, otherwise they don't get counted.

cvinh commented:
Hello
It appears that this module has not been migrated to version 11.0..14.0... do you know if it has moved to a another repository ?
If not I will start the migration
Thanks in advance
